lib/dialects/abstract/connection-manager.js

```diff
@@ -108,12 +109,8 @@ class ConnectionManager {
    * @returns {Promise}
    */
   async close() {
-    // Mark close of pool
-    this.getConnection = async function getConnection() {
-      throw new Error('ConnectionManager.getConnection was called after the connection manager was closed!');
-    };
-
-    return await this._onProcessExit();
+    await this._onProcessExit();
+    this.pool = null;
   }
```

Comment:
**Nitpick:** You can short circuit this with:

```js
async close() {
  if (!this.pool) {
    return;
  }

  // (...)
}
```

Though some may consider it redundant given that `_onProcessExit()` already has this.

Reply:
I had that consideration - about it being redundant code - so I didn't add it